File: src/main/java/io/junseok/mealmate/domain/memberwish/repository/WishRepository.java
package io.junseok.mealmate.domain.memberwish.repository;

import io.junseok.mealmate.domain.member.entity.Member;
import io.junseok.mealmate.domain.memberwish.entity.MemberWish;
import io.junseok.mealmate.domain.restaurant.entity.Restaurant;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface WishRepository extends JpaRepository<MemberWish,Long> {
    boolean existsByRestaurantAndMember(Restaurant restaurant, Member member);
    List<MemberWish> findAllByMember(Member member);

    Integer countByMember(Member member);
}

File: src/main/java/io/junseok/mealmate/domain/memberwish/repository/WishRepository.kt
package io.junseok.mealmate.domain.memberwish.repository;

import io.junseok.mealmate.domain.member.entity.Member
import io.junseok.mealmate.domain.memberwish.entity.MemberWish
import io.junseok.mealmate.domain.restaurant.entity.Restaurant
import org.springframework.data.jpa.repository.JpaRepository

interface WishRepository : JpaRepository<MemberWish,Long> {

    fun existsByRestaurantAndMember(restaurant: Restaurant, member: Member): Boolean
    fun findAllByMember(member: Member): List<MemberWish>
    fun countByMember(member: Member): Int
    fun deleteByRestaurantAndMember(restaurant: Restaurant,member: Member): Unit
}
